token_contracts/bwt/worker_token.py

```diff
@@ -66,6 +77,25 @@ def adminTransfer(self, _from: Address, _to: Address, _value: int, _data: bytes
             _data = b'None'
         super()._transfer(_from, _to, _value, _data)
 
+    @external
+    def distribute(self) -> None:
+        length = len(self._addresses)
+        tokens = self.totalSupply()
+        baln = self.create_interface_score(self._baln_token.get(), TokenInterface)
+        dist = baln.balanceOf(self.address)
+        for i in range(length):
+            address = self._addresses[i]
+            balance = self._balances[address]
+            if balance > 0:
+                amount = (dist * balance // tokens)
+                dist -= amount
+                tokens -= balance
```

Comment:
Wouldn't this be better expressed without the subtracts? Each user's relative distribution is:

`balance * dist // tokens`

where balance = user's current BWT balance, dist = total amount of BWT to distribute, tokens = bwt total supply?

Reply:
This is my algorithm for dust-free distributions. It reduces the total amount being distributed and the denominator of the share fraction on each iteration. Tokens in the equation should really be remaining_tokens and dist remaining_dist, so in the final iteration, with the final share equal to the remaining tokens, the last person gets 100% of the remaining funds being distributed.
Without that approach you will always be left with dust on the order of the number of shares. It is very small, but I like the exactness of this approach.